### PR TITLE
Add a pre-push hook

### PR DESCRIPTION
**Pull request by orlandohohmeier**
---

⚠️ This PR depends on changes that will get introduced with #1298 

---

Add a pre-push hook to run the unit test before pushing stuff to origin.

### CODE REVIEW

File: scripts/pre-push [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+source ${0%/*}/utils/message
+
+title "Run pre-push hook..."
+
+# Run the complete suite of tests
+npm run test -- -b
```
Comment (orlandohohmeier) [inline scripts/pre-push]: We should actually run the linting as well to ensure that we introduce errors with a rebase...

**Comment by orlandohohmeier** [comment]
**mesosphere-ci** retest this please

**Comment by wavesoft** [comment]
Looks great, I am OK with this, but it takes an awfully long time on every push. 
My concern is that sooner or later, people will start using `skip validation`...

**Comment by MatApple** [comment]
@orlandohohmeier can we close this PR now?

**Comment by orlandohohmeier** [comment]
@MatApple I'm fine with closing it, but would vote to merge it and risk that people will start skipping the validation (`--no-verify`) as running it every now and then is already better than never. 
We also have the commitment to improve the test performance which will improve the `pre-push` performance as well...

**Comment by MatApple** [comment]
I'm with you on that @orlandohohmeier - thought since https://github.com/dcos/dcos-ui/issues/1340 merged we are good. But I think it's good to have this as too. We can at least trial it to see if it isn't too disruptive.

**Comment by orlandohohmeier** [comment]
@MatApple @wavesoft There is a rather hacky way to get the base branch which we then could us to only test "changed" files..

```
base_branch=$(git show-branch | sed "s/].*//" | grep "\*" |
  grep -v "$(git rev-parse --abbrev-ref HEAD)" | head -n1 | sed "s/^.*\[//")
```

**Comment by MatApple** [comment]
Works for me! Hacky is Ok here :)

**Comment by nLight** [comment]
@orlandohohmeier let's rebase and merge it!

**Comment by mlunoe** [comment]
Rebased, tested and labelled!

**Pull request merged**